src/test/scala/com/vngrs/etl/BaseSparkSpec.scala

```diff
@@ -0,0 +1,39 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.BeforeAndAfterAll
+
+import scala.reflect.ClassTag
+import scala.util.Random
+
+/**
+  * Base Specification class which handles SparkContext creation and demolition
+  */
+abstract class BaseSparkSpec extends BaseSpec with BeforeAndAfterAll {
+  private val conf = new SparkConf().setMaster("local[*]").setAppName(generateAppName())
+  // TODO make SparkContext private but also provide a way to supply it to Pipeline implicitly
+  protected implicit val sc: SparkContext = new SparkContext(conf)
```

Comment:
I totally missed it. It is because Extractor is lazy but Pipeline is strict. One solution could be to create a method that takes a function `SparkContext => Pipeline[A]` and returns `Pipeline[A]`.

```scala
def createPipeline[A](f: SparkContext => Pipeline[A]): Pipeline[A] = f(sc)

val pipeline = createPipeline { implicit sc =>
  Pipeline.methodThatTakesImplicitSC()
}
```